mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -907,6 +910,27 @@ public enum ShutdownOption : int {
 		SHUT_RDWR = 0x03,   /* No more receptions or transmissions. */
 	}
 
+	// Used by libMonoPosixHelper to distinguish between different sockaddr types
+	[Map]
+	enum SockaddrType : int {
+		Invalid,
+		SockaddrStorage,
+		SockaddrUn,
+		Sockaddr,
+		SockaddrIn,
+		SockaddrIn6,
+
+		// Flag to indicate that this Sockaddr must be wrapped with a _SockaddrDynamic wrapper
+		MustBeWrapped = 0x8000,
```

Comment:
Unless you intend for "normal" people to use this value -- and I suspect you don't -- then please *remove* it. Instead of using `SockaddrType`, you can do

    partial class Sockaddr {
        internal const SockaddrType MustBeWrapped = (SockaddrType) 0x8000;
    }

This only slightly alters your code -- instead of `SockaddrType.Sockaddr | SockaddrType.MustBeWrapped`, you'd need `SockaddrType.Sockaddr | Sockaddr.MustBeWrapped`.

The benefit is that internal code is kept internal, instead of leaking into public API.